Jenkinsfile

```diff
@@ -1,136 +1,181 @@
-
-/*
- * ingeniamotion
- *
- * Copyright (c) 2020 Ingenia Motion Control.
- */
-
-
-def SW_NODE = "sw"
-def ECAT_NODE = "ecat-test-slave"
-def CAN_NODE = "canopen-test-slave"
-def BRANCH_NAME_RELEASE = "release"
-def BRANCH_NAME_DEVELOP = "develop"
-def BRANCH_NAME_MASTER = "master"
-
-if (env.BRANCH_NAME == BRANCH_NAME_DEVELOP ||
-env.BRANCH_NAME.contains(BRANCH_NAME_RELEASE) ||
-(env.CHANGE_ID && env.BRANCH_NAME.startsWith("PR-") &&
-(env.CHANGE_TARGET == BRANCH_NAME_DEVELOP || env.CHANGE_TARGET.contains(BRANCH_NAME_RELEASE)))){
-    lock('test_execution_lock_ecat') {
-        node(ECAT_NODE) {
-            deleteDir()
-
-            stage('Checkout') {
-                checkout scm
+def SW_NODE = "windows-slave"
+def ECAT_NODE = "ecat-test"
+def ECAT_NODE_LOCK = "test_execution_lock_ecat"
+def CAN_NODE = "canopen-test"
+def CAN_NODE_LOCK = "test_execution_lock_can"
+
+
+pipeline {
+    agent none
+    stages {
+        stage('Build wheels and documentation') {
+            agent {
+                docker {
+                    label SW_NODE
+                    image 'ingeniacontainers.azurecr.io/ingeniamotion-builder'
+                }
             }
-
-            stage('Install deps') {
-                bat '''
-                    python -m venv venv
-                    venv\\Scripts\\python.exe -m pip install -r requirements\\test-requirements.txt
-                '''
-            }
-
-            stage('Update FW to drives') {
-                bat """
-                    venv\\Scripts\\python.exe tests\\load_FWs.py soem
-                """
-            }
-
-            stage('Run EtherCAT embedded tests') {
-                bat '''
-                    venv\\Scripts\\python.exe -m pytest tests --protocol soem --slave 0 --html=pytest_ecat_slave_0_report.html --self-contained-html
-                    venv\\Scripts\\python.exe -m pytest tests --protocol soem --slave 1 --html=pytest_ecat_slave_1_report.html --self-contained-html
-                    exit /b 0
-                '''
-            }
-
-            stage('Save test results') {
-                archiveArtifacts artifacts: '*.html'
+            stages {
+                stage('Clone repository') {
+                    steps {
+                        bat """
+                            cd C:\\Users\\ContainerAdministrator
+                            git clone https://github.com/ingeniamc/ingeniamotion.git
+                            cd ingeniamotion
+                            git checkout ${env.GIT_COMMIT}
+                        """
+                    }
+                 }
+                stage('Install deps') {
+                    steps {
+                        bat '''
+                            cd C:\\Users\\ContainerAdministrator\\ingeniamotion
```

Comment:
Why do you go to the repository directory at each step? The workspace should already be in the correct directory

Reply:
Because we are working with the cloned repository within the container, the workspace is located elsewhere.

Reply:
Check the output of Jenkins build # 17 of the branch test-INGM-260.